zerver/views/streams.py

```diff
@@ -529,6 +543,80 @@ def you_were_just_subscribed_message(
     return message
 
 
+def you_were_just_unsubscribed_message(
+    acting_user: UserProfile,
+    recipient_user: UserProfile,
+    stream_names: Set[str],
+) -> str:
+    subscriptions = sorted(stream_names)
+    if len(subscriptions) == 1:
+        with override_language(recipient_user.default_language):
+            return _("{user_full_name} unsubscribed you from the stream {stream_name}.").format(
+                user_full_name=silent_mention_syntax_for_user(acting_user),
+                stream_name=f"#**{subscriptions[0]}**",
+            )
+
+    with override_language(recipient_user.default_language):
+        message = _("{user_full_name} unsubscribed you from the following streams:").format(
+            user_full_name=silent_mention_syntax_for_user(acting_user),
+        )
+    message += "\n\n"
+    for stream_name in subscriptions:
+        message += f"* #**{stream_name}**\n"
+    return message
+
+
+def send_messages_for_removed_subscribers(
+    removed_subscribers: List[Tuple[UserProfile, Stream]], acting_user: UserProfile
+) -> None:
+    if len(removed_subscribers) == 0:
+        return
+
+    user_id_to_unsubscribed_streams: Dict[int, Set[str]] = {}
+    user_id_to_user: Dict[int, UserProfile] = {}
+    for user, stream in removed_subscribers:
+        if user.id in user_id_to_unsubscribed_streams:
+            user_id_to_unsubscribed_streams[user.id].add(stream.name)
+        else:
+            user_id_to_unsubscribed_streams[user.id] = {stream.name}
+            user_id_to_user[user.id] = user
```

Comment:
You can simplify this code using `defaultdict(set)` for as the initial value for `user_id_to_unsubscribed_streams`.